doc/api/process.md

```diff
@@ -1177,6 +1177,44 @@ a code.
 Specifying a code to [`process.exit(code)`][`process.exit()`] will override any
 previous setting of `process.exitCode`.
 
+## process.exitWithException(error[, fromPromise])
+<!-- YAML
+added: REPLACEME
+-->
+
+* `error` {any}
+* `fromPromise` {boolean}
+
+Exit with the provided error value as if it was thrown from its original
+context. This allows an error to re-enter Node.js's error logic unmodified,
+and will use the default formatted error printing.
+
+If an error stack is available on the value, this method of exiting does not add
+additional context, unlike re-throwing using [`throw`][].
+
+This method is particularly useful when listening to the
+[`'unhandledRejection'`][] process event.
+
+As an example:
+```js
+process.on('unhandledRejection', (err) => {
```

Comment:
What happens if this is `uncaughtException`? I think that results in a dead loop? I think we should at least document about this footgun, but maybe a better solution is to just implement this separately and ignore the return value of `process._fatalException()` - or otherwise this does not exit unconditionally any more.

Reply:
Hmm, throw seems to be short-cut somehow, but this blows the stack size. :/

Reply:
Are we even supposed to emit `'uncaughtException'` when this is called? If so is there any point of using this instead of...well, just throwing an uncaught exception?